src/components/element-slot/element-slot.tsx

```diff
@@ -0,0 +1,133 @@
+import { Color } from '../colors';
+import { ElementAnchors } from '../element';
+import { Icon, IconName, IconSize } from '../icons';
+import * as React from 'react';
+import { getSpace, SpaceSize } from '../space';
+import styled from 'styled-components';
+import { tag } from '../tag';
+
+export enum ElementSlotState {
+	Default = 'default',
+	Disabled = 'disabled',
+	Highlighted = 'highlighted'
+}
+
+export interface ElementSlotProps {
+	children?: React.ReactNode;
+	description: string;
+	id: string;
+	open: boolean;
+	state: ElementSlotState;
+	title: string;
+}
+
+interface StyledElementSlotLabelProps {
+	state: ElementSlotState;
+}
+
+interface StyledIconProps {
+	open: boolean;
+}
+
+const StyledElementSlot = styled.div`
+	position: relative;
+	z-index: 1;
+`;
+
+const div = tag('div').omit(['active', 'highlight']);
+
+const LABEL_COLOR = (props: StyledElementSlotLabelProps): string => {
+	switch (props.state) {
+		case ElementSlotState.Disabled:
+			return Color.Grey60;
+		case ElementSlotState.Highlighted:
+		case ElementSlotState.Default:
+		default:
+			return 'inherit';
+	}
+};
+
+const LABEL_BACKGROUND = (props: StyledElementSlotLabelProps): string => {
+	switch (props.state) {
+		case ElementSlotState.Highlighted:
+			return Color.Grey90;
+		default:
+			return 'transparent';
+	}
+};
+
+const StyledElementSlotLabel = styled(div)`
+	position: relative;
+	display: flex;
+	align-items: center;
+	color: ${Color.Grey20};
+	position: relative;
+	font-size: 15px;
+	line-height: 21px;
+	z-index: 1;
+	color: ${LABEL_COLOR};
+	background: ${LABEL_BACKGROUND};
+`;
+
+const elementDiv = tag('div').omit(['open']);
+
+const StyledElementChildren = styled(elementDiv)`
+	flex-basis: 100%;
+	padding-left: ${getSpace(SpaceSize.L)}px;
+`;
+
+const StyledIcon = styled(Icon)`
+	position: absolute;
+	left: ${getSpace(SpaceSize.XS) + getSpace(SpaceSize.XXS)}px;
+	fill: ${Color.Grey60};
+	width: ${getSpace(SpaceSize.S)}px;
+	height: ${getSpace(SpaceSize.S)}px;
+	padding: ${getSpace(SpaceSize.XS)}px;
+	transition: transform 0.2s;
+	${(props: StyledIconProps) => (props.open ? 'transform: rotate(90deg)' : '')};
+`;
+
+const StyledElementSlotLabelContent = styled.div`
+	box-sizing: border-box;
+	margin-left: ${getSpace(SpaceSize.XXL) - 3}px;
+	overflow: hidden;
+	padding: ${getSpace(SpaceSize.XS)}px ${getSpace(SpaceSize.XXS)}px ${getSpace(SpaceSize.XS)}px
```

Comment:
this could be written shorter as `padding: ${getSpace(SpaceSize.XS)}px ${getSpace(SpaceSize.XXS)}px;`